docs/recipes/recipe-data-structures-in-git/index.md

```diff
@@ -0,0 +1,366 @@
+
+### Publish to develop workflow
+
+We would like pushes to our `develop` branch to be automatically published to our [development](/docs/managing-data-quality/testing-and-qa-workflows/) environment. Github workflows can be [triggered](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/triggering-a-workflow) by all kinds of repository events. The one we are interested in here:
+```yml
+on:
+  push:
+    branches: [develop]
+```
+
+With our trigger point worked out we need to complete a series of steps:
+1. configure snowplow-cli via environment variables provided as [github action secrets](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions)
+2. checkout our repo
+3. install snowplow-cli
+4. run the `snowplow-cli ds publish dev` command as we did earlier
+
+The full action:
+
+```yml title=".github/workflows/publish-develop.yml"
+on:
+  push:
+    branches: [develop]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      SNOWPLOW_CONSOLE_ORG_ID: ${{ secrets.SNOWPLOW_CONSOLE_ORG_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_ID: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_ID }}
+      SNOWPLOW_CONSOLE_API_KEY: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: curl -L -o snowplow-cli https://github.com/snowplow-product/snowplow-cli/releases/latest/download/snowplow-cli_linux_x86_64 && chmod u+x snowplow-cli
+
+      - run: ./snowplow-cli ds publish dev --managed-from $GITHUB_REPOSITORY
+```
+:::tip
+The value of the `--managed-from` flag will be displayed inside the 'This data structure is locked' banner we saw earlier in the UI. It is designed to help people track down the source of truth for this data structure.
+:::
+
+### Publish to production workflow
+
+In the same way we want our `develop` branch to deploy to our `develop` environment we want our `main` branch to deploy to our `production` environment.
+
+As we saw earlier publishing to production is very similar to publishing to development. The only new thing we need here is a different workflow trigger.
+
+```yml title=".github/workflows/publish-production.yml"
+on:
+  push:
+    branches: [main]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    env:
+      SNOWPLOW_CONSOLE_ORG_ID: ${{ secrets.SNOWPLOW_CONSOLE_ORG_ID }}
+      SNOWPLOW_CONSOLE_API_KEY_ID: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY_ID }}
+      SNOWPLOW_CONSOLE_API_KEY: ${{ secrets.SNOWPLOW_CONSOLE_API_KEY }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: curl -L -o snowplow-cli https://github.com/snowplow-product/snowplow-cli/releases/latest/download/snowplow-cli_linux_x86_64 && chmod u+x snowplow-cli
+
+      - run: ./snowplow-cli ds pub prod --managed-from $GITHUB_REPOSITORY
```

Comment:
`pub`? Not `published`? I assume the arg parse tool may allow that, but let's settle on `publish` as the clearer option.